src/util/cliffThoughtHeight.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Calculate the height that cliff thoughts need to extend.
+ */
+export const calculateCliffThoughtsHeight = ({ deepestDepth, depth }: { deepestDepth?: number; depth?: number }) => {
+  if (depth === undefined) return 0
+
+  // Calculate dynamic height: increase as depth decreases
+  const baseHeight = 1
+  const heightIncrease = 0.2
```

Comment:
Let's try a `heightIncrease` of `0.5` actually. `0.2` is a bit tight in practice, and we have lot of space available.